kornia/color/colormap.py

```diff
@@ -0,0 +1,183 @@
+    RGBColor(red=1, green=0.5873015873015873, blue=0),
+    RGBColor(red=1, green=0.6031746031746031, blue=0),
+    RGBColor(red=1, green=0.6190476190476191, blue=0),
+    RGBColor(red=1, green=0.6349206349206349, blue=0),
+    RGBColor(red=1, green=0.6507936507936508, blue=0),
+    RGBColor(red=1, green=0.6666666666666666, blue=0),
+    RGBColor(red=1, green=0.6825396825396826, blue=0),
+    RGBColor(red=1, green=0.6984126984126984, blue=0),
+    RGBColor(red=1, green=0.7142857142857143, blue=0),
+    RGBColor(red=1, green=0.7301587301587301, blue=0),
+    RGBColor(red=1, green=0.746031746031746, blue=0),
+    RGBColor(red=1, green=0.7619047619047619, blue=0),
+    RGBColor(red=1, green=0.7777777777777778, blue=0),
+    RGBColor(red=1, green=0.7936507936507936, blue=0),
+    RGBColor(red=1, green=0.8095238095238095, blue=0),
+    RGBColor(red=1, green=0.8253968253968254, blue=0),
+    RGBColor(red=1, green=0.8412698412698413, blue=0),
+    RGBColor(red=1, green=0.8571428571428571, blue=0),
+    RGBColor(red=1, green=0.873015873015873, blue=0),
+    RGBColor(red=1, green=0.8888888888888888, blue=0),
+    RGBColor(red=1, green=0.9047619047619048, blue=0),
+    RGBColor(red=1, green=0.9206349206349206, blue=0),
+    RGBColor(red=1, green=0.9365079365079365, blue=0),
+    RGBColor(red=1, green=0.9523809523809523, blue=0),
+    RGBColor(red=1, green=0.9682539682539683, blue=0),
+    RGBColor(red=1, green=0.9841269841269841, blue=0),
+    RGBColor(red=1, green=1, blue=0),
+]
+
+
+# Generate complete color map
+class AUTUMN(ColorMap):
+    r"""The GNU Octave colormap `autumn`"""
+
+    def __init__(self, n: int = 64, device: torch.device = torch.device('cpu'), dtype: torch.dtype = torch.float32):
+        super().__init__(_BASE_AUTUMN, n, device, dtype)
+
+
+def apply_colormap(input_tensor: Tensor, colormap: ColorMap) -> Tensor:
+    r"""Apply to a gray tensor a colormap.
+
+    Args:
+        input_tensor: the input tensor of a gray image or a batch of gray images
+        colormap: the colormap desired to be applied to the input tensor
+
+    Returns:
+        Tensor: A RGB tensor with the applied color map into the input_tensor
+
+    Example:
+        >>> input_tensor = torch.tensor([[[[0, 1, 2], [25, 50, 63]]]])
+        >>> colormap = AUTUMN()
+        >>> apply_colormap(input_tensor, colormap)
+        tensor([[[1.0000, 1.0000, 1.0000],
+                 [1.0000, 1.0000, 1.0000]],
+        <BLANKLINE>
+                [[0.0000, 0.0159, 0.0317],
+                 [0.3968, 0.7937, 1.0000]],
+        <BLANKLINE>
+                [[0.0000, 0.0000, 0.0000],
+                 [0.0000, 0.0000, 0.0000]]])
+    """
+    KORNIA_CHECK_IS_GRAY(input_tensor)
+
+    colors = colormap.get_colors()
+    keys = torch.arange(0, len(colormap), dtype=torch.int)
+
+    index = torch.bucketize(input_tensor.ravel(), keys)
+    out_tensor = colors[index].mT.reshape((3, input_tensor.shape[-2], input_tensor.shape[-1]))
```

Comment:
maybe you can stack with the proper shape in `get_colors` and avoid `mT` (which i believe is not supported by older version` + no reshape ?